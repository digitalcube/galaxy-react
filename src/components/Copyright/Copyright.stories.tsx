import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Copyright, CopyrightProps } from './Copyright'

const meta: Meta = {
  title: 'Components/Copyright',
  component: Copyright,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<CopyrightProps> = args => <Copyright {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
    serviceName: 'Shifter',
    today: new Date('2010-01-01')
}