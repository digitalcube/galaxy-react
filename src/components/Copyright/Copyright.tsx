import React, { FC } from 'react';
import { ImageDigitalcubeLogo } from '../Images';

export type CopyrightProps = {
  serviceName: string;
  today?: Date;
};
export const Copyright: FC<CopyrightProps> = ({
  serviceName,
  today = new Date(),
}) => {
  const theYear = today.getFullYear();
  return (
    <footer className="mt-auto text-center">
      <p className="m-0 common-footer-copyright">
        <small className="d-flex align-items-center justify-content-center">
          {`Â©${theYear} ${serviceName} | A Product by `}
          <a
            href="https://www.digitalcube.jp/"
            rel="nofollow noopener"
            title="DigitalCube"
          >
            <ImageDigitalcubeLogo
              className="d-block"
              width="77"
              height="15"
              alt=""
            />
          </a>
        </small>
      </p>
    </footer>
  );
};
