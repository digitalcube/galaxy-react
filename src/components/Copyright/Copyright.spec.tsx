import React from 'react';
import { render } from '@testing-library/react';
import { Default as Copyright } from './Copyright.stories';

describe('Copyright', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <Copyright
        {...{
          serviceName: 'Shifter',
          today: new Date('2010-01-01'),
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
