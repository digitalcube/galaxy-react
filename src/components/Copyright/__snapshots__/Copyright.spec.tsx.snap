// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright to match snapshot 1`] = `
<DocumentFragment>
  <p
    class="m-0 common-footer-copyright"
  >
    <small
      class="d-flex align-items-center justify-content-center"
    >
      Â©2009 Shifter | A Product by 
      <a
        href="https://www.digitalcube.jp/"
        rel="nofollow noopener"
        title="DigitalCube"
      >
        <img
          alt="digitalcube"
          class="d-block"
          height="15"
          width="77"
        />
      </a>
    </small>
  </p>
</DocumentFragment>
`;
